Added second assignment

galeal2 opened the pull request:
1) Find all ancestors of a key in a Binary Tree.
2) Find the lowest common ancestors of 2 nodes.

galeal2 committed:
Added second assignment

1) Find all ancestors of a key in a Binary Tree.
2) Find the lowest common ancestors of 2 nodes.

ariava commented on assignment_2_Q2.py:
```diff
@@ -0,0 +1,74 @@
+class Node:
```
Can you try to group the common classes in a separate library, or use assignment_2_Q1.py as a library for assignment_1_Q2.py, as you prefer? In this way you ensure that your code is what is called a "single point of truth". For example, let's say you find a bug in the BST class while you are coding the Q2 file. You must remember afterwards to replicate the fix in the Q1 file. If you avoid duplication of the code it's easier, there is only one point to change.

ariava commented on assignment_2_Q1.py:
```diff
@@ -0,0 +1,72 @@
+
+
+# find all ancestors for a Binary Tree
+def rec_find_ancestors(ptr, key, path):
```
What does "rec" stand for? Could you please use a more obvious term for the slow readers like me, or explain in a comment?

Also, can you write a short docstring that explains what the parameters are for?

ariava commented on assignment_2_Q1.py:
```diff
@@ -0,0 +1,72 @@
+class Node:
+    def __init__(self, item, left = None, right = None):
+        self.item = item
```
This is just a nit: wouldn't it be clearer to call it "key"?

ariava commented on assignment_2_Q1.py:
```diff
@@ -0,0 +1,72 @@
+# tests
+def main():
+    
+    tree = BST([16, 9, 18, 3, 14, 19, 1, 5]) 
```
Great job with testing!

ariava reviewed (changes requested):
Great job with the testing!

galeal2 committed:
Update assignment_2_Q1.py

used library, changed variable and function names and included docstring

galeal2 committed:
used libraries

galeal2 committed:
created files for the Node and BST classes

galeal2 committed:
Update assignment_2_Q2.py

add docstring

galeal2 commented:
I changed some variables' names, added docstring (i am not sure if that is right, I have never used it before) and I created different files for the Node class and BST class.
Thank you

ariava reviewed (approved):
Looks great! Well done!
